src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/CollectionExpressionTypeKind.cs

```diff
@@ -11,7 +11,8 @@ internal enum CollectionExpressionTypeKind
         Span,
         ReadOnlySpan,
         CollectionBuilder,
-        CollectionInitializer,
+        GenericIEnumerable,
+        IEnumerable,
```

Comment:
Consider naming so it's clear these are interface implementations, in contrast to `ListInterface` which is just for interfaces - for instance `ImplementsIEnumerableT` and `ImplementsIEnumerable`.